src/global_distrib/mod_global_distrib_outgoing_conns_sup.erl

```diff
@@ -33,21 +33,27 @@ add_server(Server) ->
 
 -spec get_connection(Server :: jid:lserver()) -> pid().
 get_connection(Server) ->
```

Comment:
doc for this function.
Describe errors, blocking behaviour (does it wait for server to be connected too? does it return without waiting?).

Reply:
What if several processes try to connect to the same server which is not connected?
Will it trigger TCP connect for each call or will the calls be de-duplicated?
Should be in the function description.

Reply:
ok, it will block in

gen_server:call(PoolName, wait_for_connection)
and crash with lovely error no_connections. very informative, assuming that cpool does not include pool id into the error term.

Reply:
Can be fixed by https://github.com/esl/MongooseIM/pull/1750
But this place should be covered with tests.
This comment branch is out of scope of this PR.